doc/en/example/simple.rst

```diff
@@ -895,29 +895,32 @@ here is a little example implemented via a local plugin:
 
     import pytest
 
+    phase_report_key = StashKey[Dict[str, CollectReport]]()
+
 
     @pytest.hookimpl(tryfirst=True, hookwrapper=True)
     def pytest_runtest_makereport(item, call):
         # execute all other hooks to obtain the report object
         outcome = yield
         rep = outcome.get_result()
 
-        # set a report attribute for each phase of a call, which can
+        # store test results for each phase of a call, which can
         # be "setup", "call", "teardown"
-
-        setattr(item, "rep_" + rep.when, rep)
+        item.stash.setdefault(phase_report_key, {})[rep.when] = rep
 
 
     @pytest.fixture
     def something(request):
         yield
         # request.node is an "item" because we use the default
         # "function" scope
-        if request.node.rep_setup.failed:
-            print("setting up a test failed!", request.node.nodeid)
-        elif request.node.rep_setup.passed:
-            if request.node.rep_call.failed:
-                print("executing test failed", request.node.nodeid)
+        report = request.node.stash[phase_report_key]
+        if ("setup" not in report) or report["setup"].failed:
```

Comment:
Generally I don't think it's possible to reach here when `"setup" not in report` is true (but didn't test it), so I'd remove it. Same for `"call" not in report` (as long as it's in an `elif`). Regarding `"teardown" not in report` -- it's either never true or always true, but I'm not sure which one it is off hand.

Reply:
This is probably correct on `setup` and `teardown` (tested that `teadown` doesn't exist here).

But `call` may not exist like this https://github.com/pytest-dev/pytest/pull/10442#discussion_r1023643840 no?